src/components/ListItem.jsx

```diff
@@ -41,3 +39,41 @@ export function ListItem({ name, data, listToken }) {
 		</>
 	);
 }
+
+function createNextData(data) {
```

Comment:
Great job with this function. Well crafted and easy to follow logic for a complex operation.